onnx/defs/__init__.py

```diff
@@ -120,3 +121,15 @@ def get_function_ops() -> List[OpSchema]:
 
 
 SchemaError = C.SchemaError
+
+def register_schema(op: OpSchema):
+    name: str = op.name
+    domain: str = op.domain
+    ver: int = op.since_version
+    assert ver > 0, f'OpSchema {domain}::{name} need positive version but got {ver}'
+    if has(name, domain):
+        exist_op = get_schema(name, ver, domain)
+        if exist_op is not None:
+            if exist_op.since_version == ver:
+                raise SchemaError(f'OpSchema {domain}::{name} already exist in file: {exist_op.file}:{exist_op.line}')
```

Comment:
What happens if a user registers a schema twice? Do all existing schemas have file and line info? Potentially display the info only if they exist?

Reply:
The default location of schema is set as `unknown:0`. Perhaps it would be a good idea to always display this information, particularly when it's explicitly set to this value by the user.